packages/core/utils/lib/convert-query-params.js

```diff
@@ -0,0 +1,236 @@
+'use strict';
+
+/**
+ * Converts the standard Strapi REST query params to a moe usable format for querying
+ * You can read more here: https://strapi.io/documentation/developer-docs/latest/developer-resources/content-api/content-api.html#filters
+ */
+
+const _ = require('lodash');
+
+// const BOOLEAN_OPERATORS = ['or', 'and'];
+const QUERY_OPERATORS = ['_where', '_or', '_and'];
+
+class InvalidOrderError extends Error {
+  constructor() {
+    super();
+    this.message = 'Invalid order. order can only be one of asc|desc|ASC|DESC';
+  }
+}
+class InvalidSortError extends Error {
+  constructor() {
+    super();
+    this.message =
+      'Invalid sort parameter. Expected a string, an array of strings, a sort object or an array of sort object';
+  }
+}
+
+const validateOrder = order => {
+  if (!['asc', 'desc'].includes(order.toLocaleLowerCase())) {
+    throw new InvalidOrderError();
+  }
+};
+
+/**
+ * Sort query parser
+ * @param {string} sortQuery - ex: id:asc,price:desc
+ */
+const convertSortQueryParams = sortQuery => {
```

Comment:
The return value is not always the same format ? sometimes an array and sometimes an object ?

Reply:
Yes the db layer can handle multiple formats